torch/_inductor/codecache.py

```diff
@@ -146,11 +147,81 @@ def is_gcc():
     return re.search(r"(gcc|g\+\+)", cpp_compiler())
 
 
+class _SupportedVecIsa(enum.Enum):
+    AVX512 = 1
+    AVX2 = 2
+    INVALID = -1
+
+    def __bool__(self):
+        return self != _SupportedVecIsa.INVALID
+
+    @staticmethod
+    def isa_str(supported_isa: enum.Enum):
+        if supported_isa == _SupportedVecIsa.AVX512:
+            return "avx512"
+        elif supported_isa == _SupportedVecIsa.AVX2:
+            return "avx2"
+        else:
+            return ""
+
+    @staticmethod
+    def vec_macro(supported_isa: enum.Enum):
+        if supported_isa == _SupportedVecIsa.AVX512:
+            return "CPU_CAPABILITY_AVX512"
+        elif supported_isa == _SupportedVecIsa.AVX2:
+            return "CPU_CAPABILITY_AVX2"
+        else:
+            return ""
+
+
+# Cache the cpuinfo to avoid I/O overhead. Meanwhile, the cpuinfo content
+# might have too much redundant content that is useless for ISA check. Hence,
+# we only cache some key isa information.
+@functools.lru_cache(1)
```

Comment:
```suggestion
@functools.lru_cache(None)
```

Reply:
`None` means no eviction. But system information is fixed, which means we only cache 1 item in lru_cache. May I know why `None` is better than `1`?

Reply:
Better perf, check this: https://comp.lang.python.narkive.com/pPpwxAjI/lru-cache-on-functions-with-no-arguments